src/Processors/QueryPlan/ReadFromMergeTree.cpp

```diff
@@ -1018,16 +1018,17 @@ MergeTreeDataSelectAnalysisResultPtr ReadFromMergeTree::selectRangesToRead(
     return std::make_shared<MergeTreeDataSelectAnalysisResult>(MergeTreeDataSelectAnalysisResult{.result = std::move(result)});
 }
 
-void ReadFromMergeTree::setQueryInfoInputOrderInfo(InputOrderInfoPtr order_info)
+void ReadFromMergeTree::requestReadingInOrder(size_t prefix_size, int direction, size_t limit)
 {
+    /// if dirction is not set, use current one
+    if (!direction)
+        direction = getSortDirection();
+
+    auto order_info = std::make_shared<InputOrderInfo>(SortDescription{}, prefix_size, direction, limit);
```

Comment:
let's construct it with the actual sort_description

Reply:
What's the point? It's not used here

Reply:
because it doesn't seem right. probably not important

Reply:
In general, it looks like we shouldn't use `InputOrderInfo` in `ReadFromMergeTree`. Probably, @CurtizJ has some thoughts?